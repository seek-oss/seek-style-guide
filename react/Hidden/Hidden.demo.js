import React from 'react';
import Hidden from './Hidden';
import Text from '../Text/Text';

export default {
  route: '/hidden',
  title: 'Hidden',
  category: 'Layout',
  component: Hidden,
  initialProps: {
    children: <Text strong>This content may or may not be hidden.</Text>
  },
  options: [
    {
      label: 'States',
      type: 'checklist',
      states: [
        {
          label: 'Mobile',
          transformProps: props => ({
            ...props,
            mobile: true
          })
        },
        {
          label: 'Desktop',
          transformProps: props => ({
            ...props,
            desktop: true
          })
        },
        {
          label: 'Print',
          transformProps: props => ({
            ...props,
            print: true
          })
        }
      ]
    }
  ]
};
