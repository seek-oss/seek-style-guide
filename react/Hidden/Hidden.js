import React, { PropTypes } from 'react';
import classNames from 'classnames';

import styles from './Hidden.less';

const Hidden = ({ children, className, print, screen }) => (
  <span
    className={classNames(
      className,
      {
        [styles.print]: print,
        [styles.screen]: screen
      }
    )}>{children}</span>
);

Hidden.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  print: PropTypes.bool,
  screen: PropTypes.bool
};

Hidden.defaultProps = {
  className: '',
  print: false,
  screen: false
};

export default Hidden;
