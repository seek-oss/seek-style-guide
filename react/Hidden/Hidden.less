// Todo: add responsive breakpoints support later in the same component

.print {
  @media print {
    display: none;
  }
}

.screen {
  @media screen {
    display: none;
  }
}
