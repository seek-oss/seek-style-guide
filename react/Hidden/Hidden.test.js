import React from 'react';
import renderer from 'react-test-renderer';

import Hidden from './Hidden';

describe('components/grid/hidden', () => {
  it('should render with children', () => {
    expect(
      renderer.create(
        <Hidden>Text</Hidden>
      ).toJSON()
    ).toMatchSnapshot();
  });

  it('should render with classname', () => {
    expect(
      renderer.create(
        <Hidden className="foo">Text</Hidden>
      ).toJSON()
    ).toMatchSnapshot();
  });

  it('should render with screen', () => {
    expect(
      renderer.create(
        <Hidden screen>Text</Hidden>
      ).toJSON()
    ).toMatchSnapshot();
  });

  it('should render with print', () => {
    expect(
      renderer.create(
        <Hidden print>Text</Hidden>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
