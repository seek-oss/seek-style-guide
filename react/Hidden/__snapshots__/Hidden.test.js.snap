// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grid/hidden should render with children 1`] = `
<span
  className=""
>
  Text
</span>
`;

exports[`components/grid/hidden should render with classname 1`] = `
<span
  className="foo"
>
  Text
</span>
`;

exports[`components/grid/hidden should render with print 1`] = `
<span
  className="print"
>
  Text
</span>
`;

exports[`components/grid/hidden should render with screen 1`] = `
<span
  className="screen"
>
  Text
</span>
`;
